### PR TITLE
Re-implement functional tests

### PR DESCRIPTION
**Pull request by jawid-h**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,14 +4,15 @@
   "description": "The JavaScript implementation of the Dash Platform Protocol",
   "scripts": {
     "lint": "eslint .",
-    "test": "npm run test:coverage && npm run test:browsers",
+    "test": "(EXIT_CODE=0; npm run test:coverage || EXIT_CODE=$?; npm run test:browsers || EXIT_CODE=$?; npm run test:functional || EXIT_CODE=$?; exit $EXIT_CODE)",
```
Comment (denlb) [inline package.json]: why do you need `EXIT_CODE=0;` here? looks hacky

Comment (jawid-h) [reply package.json]: We want to report failure in the end so CI build fails, yet run all possible scripts/tests.

**Comment by jawid-h** [comment]
@denlb We use term `functional` when we test bits of functionality as a black box. It's not a full scenario from a user perspective, that we call `e2e`.

File: test/functional/validateStateTransition.spec.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+      request.setStateTransition(stateTransition.serialize());
+      request.setBlockHeight(blockHeight);
+      request.setBlockHash(blockHash);
+
+      await driveUpdateStateApi.applyStateTransition(request);
+    });
+
+    const documentsStateTransition = new DocumentsStateTransition([indexDocument]);
+
+    await withinTransaction(async (blockHeight, blockHash) => {
```
Comment (shumkov) [inline test/functional/validateStateTransition.spec.js]: Let's check in the same transaction? In this case, we will check transaction isolation as well.

**Pull request merged**